harness/tests/determined/common/experimental/test_experiment.py

```diff
@@ -209,3 +212,23 @@ def test_unarchive_doesnt_update_local_on_rest_failure(
         raise AssertionError("bindings API call should raise an exception")
     except bindings.APIHttpError:
         assert expref.archived is None
+
+
+@mock.patch("determined.common.api.bindings.get_GetModelDef")
+def test_download_code_writes_output_to_file(
+    mock_bindings: mock.MagicMock,
+    make_expref: Callable[[int], experiment.Experiment],
+    tmp_path: os.PathLike,
+) -> None:
+    expref = make_expref(1)
+    sample_tgz_content = base64.b64encode(b"b64TgzResponse").decode()
```

Comment:
What's the encode/decode do here?

Reply:
There's still something I don't get. Why doesn't `sample_tgz_content = b"b64TgzResponse"`? Is it undergoing a transformation that this test is implicitly guaranteeing `download_code` does the inverse of?

Reply:
yeah, the REST API returns a response that's base64-encoded, so all clients need to decode it. this test is implicitly testing that that transformation happens as part of the file-writing process. for that to not happen though, the REST API would need to decode that response, and that's a bigger change out of the scope of this PR.